General post update

thecatfour opened the pull request:
### Category
- [ ] **Feature**
- [ ] **Bug Fix**
- [ ] **Testing**
- [ ] **Documentation**
- [X] **Other**

### Domain
- [X] **Frontend**
- [ ] **Backend**
- [ ] **Both**

### Description
This PR does the following:

- Adds the navbar and sidebar to create post page and detailed post page.
- Adds the post create button to the hubs page.
- Modifies how edit/delete buttons are displayed on post and comments.

### Related Issue(s)
This PR is more about small changes to existing components rather than adding additional functionality, so it does not have an official issue.

### New Additions
There are no new files.

### Modifications

- create-post-button.tsx: Modified to allow a hub id input as a default value. The hub id of the hub where the button was clicked is stored in localstorage.
- create-input-hub.tsx: Modified to allow a hub id input as a default value. The hub id is taken from localstorage, and it works even if the key does not exist.
- pages/create-post.tsx: Modified to include the navbar and sidebar.
- pages/detailed-post.tsx: Modified to include the navbar and sidebar.
- posts/post-list.tsx: Modified to simplify how user permissions are calculated.
- posts/post-summary: Modified to simplify how user permissions are calculated.
- comments/RecursiveComment.tsx: Modified to simplify how user permissions are calculated.
- comments/RecursiveCommentList.tsx: Modified to simplify how user permissions are calculated.
- comments/SingleComment: Modified to simplify how user permissions are calculated.
- hubs/HubPage.js: Added the create post button.
- sidebar.js: Added links to existing pages.

### New Dependencies
There are no new dependencies.

### Visuals
This is what the hub page looks like with the create post button:
![image](https://github.com/user-attachments/assets/a8f5fd9e-10fe-4051-bc56-d0ee277bef44)

This is what the create post page looks like with the navbar and sidebar:
![image](https://github.com/user-attachments/assets/4949930d-f2d1-482d-8bcc-9ed784f12252)

This is what a detailed post page looks like with the navbar and sidebar:
![image](https://github.com/user-attachments/assets/0ded5b45-93c3-4113-9765-5517bc0ae80d)

### Architecture of Your Changes
The changes to posts and comments basically involves the list component checking if the current user is the author or a mod in the hub. If the user is, pass the information to the components. This allows the buttons to render much faster as there are fewer API calls.

The create post button uses localstorage to store the default value hub. This allows users to make posts without selecting the hub input as it will be automatically set to the hub where they clicked the button. If the user changes the value in localstorage to a hub that does not exist or a hub that the user is not subscribed to, the input will just grab the first hub in the list of hubs that the user is subscribed to.

### Usage and Integration
The notable changes are to create post button and the sidebar. The create post button can be given a style as input and a default hub id as input. Neither of these are necessary, so nothing should break. The sidebar has been adjusted to fully display the first element and to include currently existing pages.

### Testing
Testing was done using the frontend and developer tools on a browser.

Localstorage manipulation does not change the functionality of the create post form. Invalid hub ids in localstorage will just not be included in the form.

### Mentions
@imawful should check to see if the create post button looks fine in his hub page.

thecatfour committed:
Update post-summary-detailed.tsx

Improved CSS formatting.

thecatfour committed:
Update page.js

thecatfour committed:
Update create-post-form.module.css

thecatfour committed:
Update create-post-form.tsx

thecatfour committed:
Update edit-post-form.module.css

thecatfour committed:
Post pages now show the sidebar and navbar

Create post page and detailed post pages now show the sidebar and navbar components.

thecatfour committed:
Update post-pages.module.css

thecatfour committed:
Update sidebar.css

Modified sidebar CSS to have a fixed width and have a top margin that accommodates the navbar.

thecatfour committed:
Update delete-post-button.tsx

If a user is on a post that they delete, redirect them to the hub of the post.

thecatfour committed:
Update create-post-button.tsx

Can now take a hub id of where to create the post. The hub id will be the default value for the hub input.

thecatfour committed:
Update create-input-hub.tsx

Updated to allow a default hub value.

thecatfour committed:
Small changes

thecatfour committed:
Update detailed-post.tsx

thecatfour committed:
Comments change

Comment components were changed to make fewer api calls when rendering the reply/delete buttons.

thecatfour committed:
Removed unnecessary modules

thecatfour committed:
Update HubPage.js

Added the create post button to the hub page component.

thecatfour committed:
Update create-post-button.tsx

Create post button can now take a hub id to set the hub input's default value. It can also take a style for the button.

thecatfour committed:
Post changes to make fewer API calls

Posts make fewer api calls when determining to render the edit/delete post buttons. It should be faster overall.

thecatfour committed:
Update sidebar.js

Updated sidebar to include profile, calendar, explore, and hubs main page.

thecatfour committed:
Update create-post-button.tsx

Changed text in button.

thecatfour committed:
Merge branch 'master' of https://github.com/UNLV-CS472-672/2024-F-GROUP3-RebelHub into general-post-update

misherwa commented:
Looks good and I'm not finding any issues. 

One thing to consider for either this PR or a different one is the color layout. This is based on the user study where the participants felt like the posts page used few colors that made the page not blend well with the other pages. So it would be a good idea to eventually add maybe more shades of grey.

A good example of a page that they liked was the Calander page, or even the login page, as it used colors from UNLV associated websites, such as Canvas or myUNLV.

thecatfour committed:
CSS changes on post pages

Post pages now have a background for the sidebar and a background for the post components.

thecatfour committed:
Merge branch 'UNLV-CS472-672:master' into general-post-update
